### ISSUE TITLE
postinstall does not run as expected on ubuntu

### ISSUE THREAD

**Issue by guysegal** [wix/react-native-interactable#146]
The `postinstall` script basically creates the symlinks:
android -> lib/android
ios -> lib/ios

This is the result of running it in OSX: 

<img width="1035" alt="osx-works" src="https://user-images.githubusercontent.com/7164560/29921670-2c0b2d0a-8e5b-11e7-8c66-3ae36ebacfad.png">

This is the result of running it in Ubuntu:

<img width="1041" alt="unbunto-problem" src="https://user-images.githubusercontent.com/7164560/29921793-b11fc28a-8e5b-11e7-972e-8ed96376f29a.png">

Trying to navigate in Ubuntu to `ios` or `android` library will not work.
<br>
It's important to mention that running `ln -sf lib/{ios,android} .` directly will work as expected:

<img width="980" alt="ubuntu-work" src="https://user-images.githubusercontent.com/7164560/29921908-27e82556-8e5c-11e7-8b15-a9fc6822a187.png">

**Comment by erezrokah** [comment]
I had the same issue.
The problem is that the postinstall script uses a specific bash feature called expansion, but by default npm uses DASH and not BASH:
https://github.com/travis-ci/travis-ci/issues/3963

A temporary fix is to set npm script-shell using:
`npm config set script-shell "/bin/bash"` or something similar

**Comment by rotemmiz** [comment]
The current postinstall script does not support Windows, and is creating multiple issues for users

@silyevsk , what do you think about reimplementing it using https://nodejs.org/api/fs.html#fs_fs_symlink_target_path_type_callback ?

**Comment by sondremare** [comment]
Anything happening here? It's breaking my build aswell

**Comment by chrismllr** [comment]
I am also getting a breaking Android build in `circleci`:
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not list contents of '/home/circleci/android/node_modules/react-native-interactable/{ios,android}'. Couldn't follow symbolic link.
```

**Comment by chrismllr** [comment]
The above PR does the trick for me, but I haven't tried on a Windows machine.

**Comment by erezrokah** [comment]
@chrismllr I posted the pr as a quick fix for Linux environments and I'm not sure it will work for Windows machines (unless you have `bash` installed).
Anyway it is much better than the current state of having the script work only Mac machines.

**Comment by chrismllr** [comment]
@erezrokah It has been some time since this discussion, but I tried a different approach in that same PR #172 - it seems heavy handed as far as file changes go, but also seems like the most straightforward solution. Thoughts?

**Comment by helloagain-dev** [comment]
This might be a stupid question, but why don't you simply split the commands in 2?
``ln -s lib/android . && ln -s lib/ios .``

**Comment by helloagain-dev** [comment]
I would really appreciate if this gets fixed quickly, as this breaks our automated build.

**Comment by BonnieMilian** [comment]
@helloagain-dev my automated build is broken too, did you do a build in another way?

**Comment by gsaandy** [comment]
any workaround?

**Comment by helloagain-dev** [comment]
Not so far.

**Comment by gsaandy** [comment]
Manually doing the following after installing dependencies worked for me
`cd $PROJECT_DIR/node_modules/react-native-interactable`
`rm -rf \{ios\,android\} && rm -rf android && rm -rf ios`
`ln -s lib/android . && ln -s lib/ios .`

**Comment by helloagain-dev** [comment]
I don't see why the PR that fixes Linux is not getting merged.
Who has automated builds on Windows anyway?

**Comment by markhu** [comment]
Yet another possible fix: #194 ultra-simple one-liner calling `bash -c ''` might work.
Or

    $ cat .npmrc
    script-shell=/bin/bash

**Comment by andreluisjunqueira** [comment]
It's breaking my Bitrise automation too

**Comment by mkuczera** [comment]
We are currently working on maintenance work for the whole package. Please just be patient. We are fixing a bunch of issues right now and thinking about the following steps.

**Comment by isaachinman** [comment]
@mkuczera Two months later now... any update? Setting up a Bitrise workflow and found `react-native-interactable` was breaking the build.

I can't see a good reason for setting up a project like this, and along with all the people above, just need a simple solution to get my app(s) deployed.

**Comment by evanjmg** [comment]
You can use the forked version here -https://github.com/DylanVann/react-native-interactable - it's from this PR which hasn't been merged yet. -> https://github.com/wix/react-native-interactable/pull/211/files
When that PR is merged this can be closed

**Comment by st-sloth** [comment]
This issue is fixed in #211. Would be nice to get it merged and stop this torrent of installing problems.

**Comment by mathias5r** [comment]
Guys, I made a script to solve the issue with the response of @gsaandy. It's bad when you need to run this many times.

https://github.com/mathias5r/fix-react-native-interactable-ubuntu
